numba/tests/test_debuginfo.py

```diff
@@ -596,6 +596,24 @@ def foo():
             self.assertRegex(data_type, expected[field])
 
     def test_debug_optnone(self):
+        def get_debug_lines(fn):
+            # Get the lines contained in the debug info
+            mdlist = self._get_metadata(fn, fn.signatures[0])
+            lines = set()
+            for md in mdlist:
+                m = re.match(r"!\d+ = !DILocation\(line: (\d+),", md)
+                if m:
+                    ln = int(m.group(1))
+                    lines.add(ln)
+            return lines
```

Comment:
This is quite like: https://github.com/numba/numba/pull/7617/files#diff-f5c7dace4d41363f673092ed1b4030c54136136260a87cfe2916faeeac985c4fR151-R162 which is a refactor of these tests to pull out the code that extracts the metadata.

Reply:
refactored in https://github.com/numba/numba/pull/7764/commits/ece8ce1b2d96aa7914a27312a5fdb8c986f5b239